relay-statsd/src/lib.rs

```diff
@@ -162,6 +161,26 @@ pub fn set_client(client: MetricsClient) {
     *METRICS_CLIENT.write() = Some(Arc::new(client));
 }
 
+/// Set a test client for the period of the called function.
+/// Note: This also changes the METRICS_CLIENT for other threads.
+pub fn with_capturing_test_client(f: impl FnOnce()) -> Vec<String> {
+    let old_client = METRICS_CLIENT.read().clone();
+
+    let (rx, sink) = SpyMetricSink::new();
+    let test_client = MetricsClient {
+        statsd_client: StatsdClient::from_sink("", sink),
+        default_tags: Default::default(),
+        sample_rate: 1.0,
+    };
+    set_client(test_client);
+
+    f();
+
+    *METRICS_CLIENT.write() = old_client;
```

Comment:
I think you might have those changes locally, but this needs to be done per-thread